src/main/java/xyz/aadev/generitech/common/tileentities/power/TileEntityPowerStorage.java

```diff
@@ -0,0 +1,188 @@
+ */
+
+import net.darkhax.tesla.api.ITeslaConsumer;
+import net.darkhax.tesla.api.ITeslaHolder;
+import net.darkhax.tesla.api.implementation.BaseTeslaContainer;
+import net.darkhax.tesla.capability.TeslaCapabilities;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.inventory.IInventory;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.ITickable;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraftforge.common.capabilities.Capability;
+import xyz.aadev.aalib.common.inventory.InternalInventory;
+import xyz.aadev.aalib.common.inventory.InventoryOperation;
+import xyz.aadev.generitech.client.gui.power.GuiGenerator;
+import xyz.aadev.generitech.client.gui.power.GuiPowerStorage;
+import xyz.aadev.generitech.common.container.power.ContanierGenerator;
+import xyz.aadev.generitech.common.container.power.ContanierPowerStorage;
+import xyz.aadev.generitech.common.tileentities.TileEntityMachineBase;
+
+import javax.annotation.Nullable;
+
+public class TileEntityPowerStorage extends TileEntityMachineBase implements ITeslaHolder, ITeslaConsumer, ITickable {
+    private InternalInventory inventory = new InternalInventory(this, 1);
+    private BaseTeslaContainer container = new BaseTeslaContainer(5000, 50000, 1000, 1000);
+    private int[] sides = new int[6];
+
+
+
+
+
+    @Override
+    public void update() {
+        BlockPos pos = getPos();
+        World worldIn = getWorld();
+
+    if (worldIn.isBlockPowered(pos)){
+        sides[0]=2;
+    }else {
+        sides[0]=0;
+    }
+        distributepowertoface.transferPower(pos,worldIn,120,container,sides);
+
+    }
+
+
+
+
+
+
+    @Override
+    public void readFromNBT(NBTTagCompound nbtTagCompound) {
+        super.readFromNBT(nbtTagCompound);
+
+        this.container = new BaseTeslaContainer(nbtTagCompound.getCompoundTag("TeslaContainer"));
+        this.sides = nbtTagCompound.getIntArray("sides");
+
+
+    }
+
+
+    @Override
+    public NBTTagCompound writeToNBT(NBTTagCompound nbtTagCompound) {
+        super.writeToNBT(nbtTagCompound);
+        nbtTagCompound.setTag("TeslaContainer", this.container.serializeNBT());
+        nbtTagCompound.setIntArray("sides",sides);
+
+
+        return nbtTagCompound;
+    }
+
+
+
+    @Override
+    public IInventory getInternalInventory() {
+        return inventory;
+    }
+
+    @Override
+    public void onChangeInventory(IInventory inv, int slot, InventoryOperation operation, ItemStack removed, ItemStack added) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AS1186)